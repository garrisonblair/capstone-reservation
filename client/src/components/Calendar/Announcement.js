import React, { Component } from 'react';
import { Message } from 'semantic-ui-react';
import moment from 'moment';
import api from '../../utils/api';
import './Calendar.scss';

class Announcement extends Component {
  state = {
    announcements: [],
    visible: true,
  }

  componentDidMount() {
    api.getAllAnnouncements()
      .then((response) => {
        this.setState({ announcements: response.data });
      });
  }

  handleDismiss = () => {
    this.setState({ visible: false });
  }

  render() {
    const { visible } = this.state;
    if (!visible) {
      return null;
    }
    const { announcements } = this.state;
    const currentDate = moment().format('YYYY-MM-DD');
    const text = [];
    announcements.forEach((a) => {
      if (moment(currentDate).isBetween(a.start_date, a.end_date)
      || moment(currentDate).isSame(a.start_date)
      || moment(currentDate).isSame(a.end_date)) {
        text.push(
          <Message.Item>
            <b>
              {a.title}
            </b>
            :&nbsp;
            {a.content}
          </Message.Item>,
        );
      }
    });
    if (text === []) {
      return null;
    }
    return (
      <Message
        onDismiss={this.handleDismiss}
        negative
        className="announcement"
      >
        <Message.List>
          {text}
        </Message.List>
      </Message>
    );
  }
}

export default Announcement;
