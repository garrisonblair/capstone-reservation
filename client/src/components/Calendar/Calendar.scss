/* Variables */

:root {
  --rowNum: 7;
  --colNum: 5;
  --gridHeight: 120px;
  --wrapperH: 40px;
  --cellsDivisionNum: 0;
  --opacity: 1;
  --pointerEvents: auto;
}

$textColor: black;
$textColorHeader: white;
$borderColor: #F2D692;
$eventBackgroundColor: aqua;
$backgroundColorContainer: #F2D692;
$backgroundColorCells: #C9A859;
$backgroundColorRooms: #A17F2E;
$backgroundColorHours: #A17F2E;
$borderWidth: 1px;
$cellsMinHeight: auto;

/* Calendar */

body {
  overflow: scroll;
}

.calendar__container {
  opacity: var(--opacity);
  pointer-events: var(--pointerEvents);
  position: fixed;
  display: inline-grid;
  margin-top: 65px;
  // padding: 20px;
  width: 100%;
  height: 100%;
  overflow: scroll;
  overflow-x: visible;
  background-color: $backgroundColorContainer;
  grid-template-rows: max-content auto;
  padding-bottom: 100px;
}

.calendar__wrapper {
  display: inline-grid;
  padding: 2px;
  height: 100%;
  grid-template-rows: max-content auto;
}

.date__container {
  position: relative;
  width: 100%;
  margin: 0;
}

.calendar__date {
  color: white;
  margin-top: 10px;
  font-size: 50;
  &__header {
    position: -webkit-sticky;
    position: sticky;
    left: 40%;
    display: inline;
    font-weight: 400;
    margin: 0 10px;
  }
}

.calendar__rooms__wrapper {
  display: inline-grid;
  position: -webkit-sticky;
  position: sticky;
  // height: 50px;
  color: $textColorHeader;
}

.calendar__hours__wrapper {
  display: inline-grid;
  position: -webkit-sticky;
  position: sticky;
  color: $textColorHeader;
}

.calendar__cells__wrapper {
  display: inline-grid;
  height: 100%;
}

.calendar__cells__cell {
  border: $borderWidth solid $borderColor;
  margin-top: -$borderWidth;
  margin-left: -$borderWidth;
  min-height: $cellsMinHeight;
  width: 120px;
  // height: 50px;
  background-color: $backgroundColorCells;
  &--campon {
    font-size: 9;
  }
  border-radius: 5px;
}

.calendar__cells__cell:hover {
  box-shadow: 0 0 20px #A17F2E;
  cursor: pointer;
}

.calendar__booking {
  padding: 1px;
  padding-top: 3px;
  font-size: 12px;
  color: white;
  font-weight: 100;
  border-radius: 5px;
  span {
    line-height: 10px;
  }
}

.calendar__booking:hover {
  box-shadow: 0 0 20px rgba(33, 33, 33, .2);
  cursor: pointer;
}

.calendar__rooms__cells {
  display: inline-grid;
}

.calendar__rooms__room {
  display: flex;
  justify-content: center;
  align-items: center;
  border: $borderWidth solid $borderColor;
  padding: 5px 10px;
  margin-top: -$borderWidth;
  margin-left: -$borderWidth;
  border-radius: 5px;
  background-color: $backgroundColorRooms;
}

.calendar__hours__hour {
  display: flex;
  justify-content: center;
  align-items: center;
  border: $borderWidth solid $borderColor;
  margin-top: -$borderWidth;
  margin-left: -$borderWidth;
  // padding: 0 1px;
  padding: 5px 10px;
  border-radius: 5px;
  background-color: $backgroundColorHours;
}

.modals {
  display: none;
}

.menu--date:hover {
  cursor: pointer;
}

.menu--date:before {
  content: none !important;
}

.arrow {
  color: #932439 !important;
  // background-color: #4D4D4D !important;
  margin-left: 5px !important;
}

.arrow:hover {
  background-color: rgb(87, 85, 85);
}

.button_orientation {
  border: 2px solid #932439;
  border-radius: 5px;
  background-color: $backgroundColorHours;
  width: 98.5%;
  height: 95%;
  color: white;
  text-align: center;
  line-height: 100%;
}

.button_orientation:hover {
  background-color: rgb(124, 98, 37);
  cursor: pointer;
}