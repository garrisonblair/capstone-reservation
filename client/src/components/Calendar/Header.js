import React, { Component } from 'react';
import './Calendar.scss';


class Header extends Component {
  state = {
    headerList: [],
    headerNum: 0,
    type: 'column',
    name: 'room',
  }

  static getDerivedStateFromProps(props) {
    return {
      headerList: props.headerList,
      headerNum: props.headerList.length,
      type: props.type,
      name: props.name,
    };
  }

  setStyle() {
    const { headerNum, type } = this.state;
    let style;
    if (type === 'column') {
      style = {
        cell: {
          minWidth: '120px',
        },
        wrapper: {
          gridTemplateColumns: `repeat(${headerNum}, 1fr)`,
          gridColumnStart: 2,
          top: 0,
        },
      };
    } else {
      style = {
        cell: {
          minHeight: '60px',
          width: '90px',
        },
        wrapper: {
          gridTemplateRows: `repeat(${headerNum}, 1fr)`,
          left: 0,
        },
      };
    }
    return style;
  }

  render() {
    const { headerList, name } = this.state;
    const header = headerList.map(h => (
      <div className={`calendar__${name}s__${name}`} style={this.setStyle().cell} key={h.name ? h.name : h}>
        {h.name ? h.name : h}
      </div>
    ));

    return <div className={`calendar__${name}s__wrapper`} style={this.setStyle().wrapper}>{header}</div>;
  }
}

export default Header;
