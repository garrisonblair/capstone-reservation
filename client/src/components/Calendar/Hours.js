import PropTypes from 'prop-types';
import React, {Component} from 'react';
import './Calendar.scss';


class Hours extends Component {

  state = {
    hoursList: [],
  }

  static getDerivedStateFromProps(props, state) {
    if(props.hoursList === state.hoursList) {
      return null;
    }
    return {
      hoursList: props.hoursList,
    };
  }

  render() {
    const {hoursList} = this.state;
    const hours = hoursList.map((hour) =>
      <div className="calendar__hours__hour" key={hour}>
        {hour}
      </div>
    );

    return <div className="calendar__hours__wrapper">{hours}</div>
  }
}

Hours.propTypes = {
  hoursList: PropTypes.array,
}


export default Hours;
