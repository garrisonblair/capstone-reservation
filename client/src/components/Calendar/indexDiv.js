import React, {Component} from 'react';
import settings from '../../config/settings';
import './Calendar.scss';


class Calendar extends Component {
  
  state = {
    roomsList: [1,2,3,4,5,6,7,8,9,10],
    dateSelected: new Date(),
    hourStart: 8,
    hourEnd: 23,
    hourIncrement: 1
  };

  componentDidMount() {
    console.log(settings)
  }

  renderHeader() {
    return (
      <div className="header row flex-middle">
        {/* <div className="col col-start">
          <div className="icon" onClick={this.prevMonth}>
            chevron_left
          </div>
        </div> */}
        <div className="col col-center">
          <span>
            {this.state.dateSelected.toDateString()}
          </span>
        </div>
        {/* <div className="col col-end" onClick={this.nextMonth}>
          <div className="icon">chevron_right</div>
        </div> */}
      </div>
    );
  }

  renderRooms() {
    const { roomsList } = this.state;
    const rooms = roomsList.map((room) =>
      <div className="col col-center" key={room}>
        {room}
      </div>
    );
    
    return <div className="calendar__rooms row">{rooms}</div>
  }

  renderHours() {
    const { hourStart, hourEnd, hourIncrement } = this.state;
    const hours = [];

    for (let hour = hourStart; hour <= hourEnd; hour += hourIncrement) {
      hours.push(
        <div className="calendar__hours--label" key={hour}>
          {hour}
        </div>
      )
    }

    return <div className="calendar__hours">{hours}</div>
  }

  renderCells() {
    const { hourStart, hourEnd, hourIncrement } = this.state;
    const { roomsList } = this.state;

    let cells = [];
    let rows = [];
    let numberOfRows = (hourEnd - hourStart + 1) / hourIncrement;
    let numberOfCells = numberOfRows * (roomsList.length);
    let cell = 0;

    while (cell < numberOfCells) {
      for (let j = 0; j < roomsList.length; j++) {
        cells.push(
          <div className={`calendar__cells__cell-${j} ${
            this.isBooked(currentRoom, currentHour)
              ? "booked"
              : ""
          }`} style={this.setCellStyle(j).cell_style} key={cell} data-hour={currentHour} data-room={currentRoom} onClick={this.handleClickCell}>{this.cellText(currentRoom, currentHour)}</div>
        );
        cell++;
      }
      
      rows.push(
        <div className="row" key={cell}>
          {cells}
        </div>
      )
      cells = [];
    }

    return <div className="body">{rows}</div>;

  }

  handleClickCell(e) {
    alert(e.target.getAttribute('data-key'));
  }
  // schedules() {
  //   axios({
  //     method: 'GET',
  //     url: `${settings.API_ROOT}/videoframes`,
  //    })
  //    .then((response) => {
  //      const schedules = response.data.map(schedule => schedule);
  //      this.setState({schedule});
  //    })
  //    .catch(error => {
  //     console.log(error);
  //     const errorMsg = "Oops, something went wrong while fetching items!";
  //     this.setState({errorMsg});
  //   })
  // }

  render() {
    return (
      <div className="calendar">
        {this.renderHeader()}
        {this.renderRooms()}
        {/* {this.renderHours()} */}
        {this.renderCells()}   
      </div>
    )
  }
}

export default Calendar;
