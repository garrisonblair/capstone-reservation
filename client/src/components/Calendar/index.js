import React, { Component } from 'react';
import { Icon } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import Cells from './Cells';
import Header from './Header';
import Navigation from '../Navigation';
import api from '../../utils/api';
import storage from '../../utils/local-storage';
import './Calendar.scss';

class Calendar extends Component {
  static timeStringToInt(time) {
    const tokens = time.split(':');
    const timeInt = {
      hour: parseInt(tokens[0], 10),
      minutes: parseInt(tokens[1], 10),
    };
    return timeInt;
  }

  static onOpenDatePicker() {
    document.documentElement.style.setProperty('--opacity', 0.2);
    document.documentElement.style.setProperty('--pointerEvents', 'none');
  }

  static onCloseDatePicker() {
    document.documentElement.style.setProperty('--opacity', 1);
    document.documentElement.style.setProperty('--pointerEvents', 'auto');
  }

  state = {
    roomsList: [],
    hoursList: [],
    selectedDate: new Date(),
    roomsNum: 0,
    hoursDivisionNum: 0,
    orientation: 1,
    update: false,
  };

  /*
   * COMPONENT LIFE CYCLE
   */

  componentDidMount() {
    document.body.style.backgroundColor = '#bdc3c7';
    this.setState(
      { orientation: storage.getOrientation() === null ? 1 : storage.getOrientation() },
    );
    this.getBookings();
    this.getRooms();
    this.setHours();
  }

  componentWillUnmount() {
    document.body.style.backgroundColor = 'white';
  }

  /*
   * REQUESTS
   */

  // propsTesting* is used for Jest testing
  getBookings() {
    const { propsTestingBookings } = this.props;
    const test = !!propsTestingBookings;
    if (test) {
      this.setState({ bookings: propsTestingBookings });
    } else {
      const { selectedDate } = this.state;
      const params = {
        year: selectedDate.getFullYear(),
        month: selectedDate.getMonth() + 1,
        day: selectedDate.getDate(),
      };

      api.getBookings(params)
        .then((response) => {
          this.getCampOns(params, response.data);
        });
    }
  }

  getCampOns(params, bookings) {
    const { propsTestingCampOns } = this.props;
    const test = !!propsTestingCampOns;
    if (test) {
      this.setState({ campOns: propsTestingCampOns });
    } else {
      api.getCampOns(params)
        .then((response) => {
          this.campOnToBooking(response.data, bookings);
        });
    }
  }

  getRooms() {
    const { propsTestingRooms } = this.props;
    const test = !!propsTestingRooms;
    if (test) {
      this.setState({ roomsList: propsTestingRooms });
    } else {
      api.getRooms()
        .then((response) => {
          this.setState({ roomsList: response.data, roomsNum: response.data.length });
        });
    }
  }

  setHours = () => {
    const hoursSettings = {
      start: '08:00',
      end: '22:00',
      increment: 60,
    };
    const hourStart = Calendar.timeStringToInt(hoursSettings.start);
    const hourEnd = Calendar.timeStringToInt(hoursSettings.end);

    const minutesIncrement = hoursSettings.increment;
    const hours = [];
    const time = new Date();
    time.setHours(hourStart.hour, hourStart.minutes, 0);

    // Format time for display in table
    let currentTime = hourStart.hour * 60 + hourStart.minutes;
    const endTime = hourEnd.hour * 60 + hourEnd.minutes;

    while (currentTime <= endTime) {
      hours.push(time.toLocaleTimeString([], { hour: '2-digit', minute: '2-digit' }));
      time.setMinutes(time.getMinutes() + minutesIncrement);
      currentTime += minutesIncrement;
    }
    const hoursDivisionNum = minutesIncrement * hours.length / 10;

    this.setState({
      hoursSettings,
      hoursList: hours,
      hoursDivisionNum,
    });
  }

  setStyle() {
    const { orientation } = this.state;
    let style;
    if (orientation === 0) {
      style = {
        wrapper: {
          gridTemplateRows: 'max-content auto',
        },
      };
    } else {
      style = {
        wrapper: {
          gridTemplateColumns: 'max-content auto',
        },
      };
    }
    return style;
  }

  /*
   * HELPER METHOD
   */

  changeDate = (date) => {
    this.setState({ selectedDate: date }, () => {
      this.getBookings();
    });
  }

  changeOrientation = () => {
    const { orientation } = this.state;
    const newOrientation = orientation === 0 ? 1 : 0;
    storage.saveOrientation(newOrientation);
    this.setState({ orientation: newOrientation });
  }

  onCloseModalWithAction = () => {
    this.getBookings();
  }

  campOnToBooking = (campOns, bookings) => {
    const campOnBookings = [];
    if (!!campOns && !!bookings) {
      campOns.forEach((campOn, index) => {
        let dateCampOn = '';
        let roomCampOn = '';
        if (bookings) {
          for (let i = 0; i < bookings.length; i += 1) {
            if (bookings[i].id === campOn.camped_on_booking) {
              const { date, room } = bookings[i];
              dateCampOn = date;
              roomCampOn = room;
              break;
            }
          }
        }
        campOnBookings.push({
          date: dateCampOn,
          start_time: campOn.start_time,
          end_time: campOn.end_time,
          booker: campOn.booker,
          room: roomCampOn,
          id: `camp${campOn.camped_on_booking}-${index}`,
          isCampOn: true,
          camped_on_booking: campOn.camped_on_booking,
        });
      });
      campOnBookings.forEach((campOnBooking) => {
        bookings.push(campOnBooking);
      });
      this.setState({ bookings, campOns });
    }
  }

  update = () => {
    this.getBookings();
    this.setState({ update: true });
    this.setState({ update: false });
  }

  /*
   * COMPONENT RENDERING
   */

  render() {
    const {
      roomsList,
      hoursList,
      hoursSettings,
      bookings,
      campOns,
      selectedDate,
      orientation,
      roomsNum,
      hoursDivisionNum,
      update,
    } = this.state;
    let colList = roomsList;
    let colName = 'room';
    let rowList = hoursList;
    let rowName = 'hour';
    if (orientation === 1) {
      rowList = roomsList;
      rowName = 'room';
      colList = hoursList;
      colName = 'hour';
    }
    return (
      <div className="main__page">
        <Navigation
          key={0}
          showDate
          changeDate={this.changeDate}
          onOpenDatePicker={Calendar.onOpenDatePicker}
          onCloseDatePicker={Calendar.onCloseDatePicker}
          update={update}
        />
        <div className="calendar__container" key={1}>
          <div className="calendar__wrapper" style={this.setStyle().wrapper}>
            <button type="button" className="button_orientation" onClick={() => this.changeOrientation()}><Icon name="exchange" /></button>
            <Header headerList={colList} type="column" name={colName} />
            <Header headerList={rowList} type="row" name={rowName} />
            <Cells
              hoursSettings={hoursSettings}
              bookings={bookings}
              roomsList={roomsList}
              hoursList={hoursList}
              campOns={campOns}
              selectedDate={selectedDate}
              onCloseModalWithAction={this.onCloseModalWithAction}
              orientation={orientation}
              roomsNum={roomsNum}
              hoursDivisionNum={hoursDivisionNum}
              update={this.update}
            />
          </div>
        </div>
      </div>
    );
  }
}

Calendar.propTypes = {
  propsTestingBookings: PropTypes.instanceOf(Object),
  propsTestingCampOns: PropTypes.instanceOf(Object),
  propsTestingRooms: PropTypes.instanceOf(Object),
};

Calendar.defaultProps = {
  propsTestingBookings: null,
  propsTestingCampOns: null,
  propsTestingRooms: null,
};

export default Calendar;
