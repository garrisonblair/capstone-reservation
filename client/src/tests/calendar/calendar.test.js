/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import 'react-dates/initialize';
import Calendar from '../../components/Calendar/index';


it('Shows calendar', () => {
  const roomsMock = [{ id: 1, name: 'room 1' }, { id: 2, name: 'room 2' }];
  const bookingsMock = [{
    id: 1,
    student: '12345567',
    room: 1,
    date: '2018-10-09',
    start_time: '08:10:00',
    end_time: '09:50:00',
    booker: {
      username: 'Ken',
    },
  }];

  const tree = renderer
    .create(
      <Router>
        <Calendar propsTestingRooms={roomsMock} propsTestingBookings={bookingsMock} />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
