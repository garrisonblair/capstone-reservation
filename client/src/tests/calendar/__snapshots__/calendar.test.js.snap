// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows calendar 1`] = `
<div
  className="main__page"
>
  <div
    className="navigation"
  >
    <div
      className="ui inverted top fixed navigation__bar menu"
    >
      <a
        className="item navigation__title"
        onClick={[Function]}
      >
        Capstone
      </a>
      <div
        className="item"
        onClick={[Function]}
      >
        <a
          href="https://docs.google.com/forms/u/1/d/1g-d02gd4s1JQjEEArGkwZVmlYcBeWlDL6M3R2dcFmY8/edit?usp=sharing"
        >
          Feedback
        </a>
      </div>
      <div
        className="right item menu--date"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="angle left circular icon arrow"
          onClick={[Function]}
        />
        <i
          aria-hidden="true"
          className="calendar alternate outline icon"
          onClick={[Function]}
        />
        <div
          className="datepicker"
        >
          <div
            className="SingleDatePicker SingleDatePicker_1"
          >
            <div>
              <div
                className="SingleDatePickerInput SingleDatePickerInput_1 SingleDatePickerInput__withBorder SingleDatePickerInput__withBorder_2"
              >
                <div
                  className="DateInput DateInput_1"
                >
                  <input
                    aria-describedby="DateInput__screen-reader-message-datepicker"
                    aria-label="Date"
                    autoComplete="off"
                    className="DateInput_input DateInput_input_1"
                    disabled={false}
                    id="datepicker"
                    name="datepicker"
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Date"
                    readOnly={false}
                    required={false}
                    type="text"
                    value="02/02/2019"
                  />
                  <p
                    className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                    id="DateInput__screen-reader-message-datepicker"
                  >
                    Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <span
          onClick={[Function]}
          onKeyDown={[Function]}
          role="presentation"
        >
          Sat Feb 2nd 2019
        </span>
        <i
          aria-hidden="true"
          className="angle right circular icon arrow"
          onClick={[Function]}
        />
      </div>
      <div
        className="right menu navigation__container"
        inverted="true"
      >
        
        
        
        <a
          className="item navigation__login"
          onClick={[Function]}
        >
          Login
        </a>
      </div>
    </div>
  </div>
  <div
    className="calendar__container"
  >
    <div
      className="calendar__wrapper"
      style={
        Object {
          "gridTemplateColumns": "max-content auto",
        }
      }
    >
      <button
        className="button_orientation"
        onClick={[Function]}
        type="button"
      >
        <i
          aria-hidden="true"
          className="exchange icon"
        />
      </button>
      <div
        className="calendar__hours__wrapper"
        style={
          Object {
            "gridColumnStart": 2,
            "gridTemplateColumns": "repeat(15, 1fr)",
            "top": 0,
          }
        }
      >
        <div
          className="calendar__hours__hour"
          style={
            Object {
              "minWidth": "120px",
            }
          }
        >
          08:00
        </div>
        <div
          className="calendar__hours__hour"
          style={
            Object {
              "minWidth": "120px",
            }
          }
        >
          09:00
        </div>
        <div
          className="calendar__hours__hour"
          style={
            Object {
              "minWidth": "120px",
            }
          }
        >
          10:00
        </div>
        <div
          className="calendar__hours__hour"
          style={
            Object {
              "minWidth": "120px",
            }
          }
        >
          11:00
        </div>
        <div
          className="calendar__hours__hour"
          style={
            Object {
              "minWidth": "120px",
            }
          }
        >
          12:00
        </div>
        <div
          className="calendar__hours__hour"
          style={
            Object {
              "minWidth": "120px",
            }
          }
        >
          13:00
        </div>
        <div
          className="calendar__hours__hour"
          style={
            Object {
              "minWidth": "120px",
            }
          }
        >
          14:00
        </div>
        <div
          className="calendar__hours__hour"
          style={
            Object {
              "minWidth": "120px",
            }
          }
        >
          15:00
        </div>
        <div
          className="calendar__hours__hour"
          style={
            Object {
              "minWidth": "120px",
            }
          }
        >
          16:00
        </div>
        <div
          className="calendar__hours__hour"
          style={
            Object {
              "minWidth": "120px",
            }
          }
        >
          17:00
        </div>
        <div
          className="calendar__hours__hour"
          style={
            Object {
              "minWidth": "120px",
            }
          }
        >
          18:00
        </div>
        <div
          className="calendar__hours__hour"
          style={
            Object {
              "minWidth": "120px",
            }
          }
        >
          19:00
        </div>
        <div
          className="calendar__hours__hour"
          style={
            Object {
              "minWidth": "120px",
            }
          }
        >
          20:00
        </div>
        <div
          className="calendar__hours__hour"
          style={
            Object {
              "minWidth": "120px",
            }
          }
        >
          21:00
        </div>
        <div
          className="calendar__hours__hour"
          style={
            Object {
              "minWidth": "120px",
            }
          }
        >
          22:00
        </div>
      </div>
      <div
        className="calendar__rooms__wrapper"
        style={
          Object {
            "gridTemplateRows": "repeat(2, 1fr)",
            "left": 0,
          }
        }
      >
        <div
          className="calendar__rooms__room"
          style={
            Object {
              "minHeight": "60px",
              "width": "90px",
            }
          }
        >
          room 1
        </div>
        <div
          className="calendar__rooms__room"
          style={
            Object {
              "minHeight": "60px",
              "width": "90px",
            }
          }
        >
          room 2
        </div>
      </div>
      <div
        className="calendar__cells__wrapper"
        style={
          Object {
            "gridTemplateRows": "repeat(0, 1fr)",
          }
        }
      >
        <div
          className="calendar__rooms__cells"
          style={
            Object {
              "gridTemplateColumns": "repeat(90, 1fr)",
            }
          }
        >
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 7,
                "gridColumnStart": 1,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 13,
                "gridColumnStart": 7,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 19,
                "gridColumnStart": 13,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 25,
                "gridColumnStart": 19,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 31,
                "gridColumnStart": 25,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 37,
                "gridColumnStart": 31,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 43,
                "gridColumnStart": 37,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 49,
                "gridColumnStart": 43,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 55,
                "gridColumnStart": 49,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 61,
                "gridColumnStart": 55,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 67,
                "gridColumnStart": 61,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 73,
                "gridColumnStart": 67,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 79,
                "gridColumnStart": 73,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 85,
                "gridColumnStart": 79,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 91,
                "gridColumnStart": 85,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__booking"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="button"
            style={
              Object {
                "backgroundColor": "#7F7F7F",
                "borderLeft": "1px solid white",
                "borderRight": "1px solid white",
                "borderTop": "1px solid white",
                "gridColumnEnd": 12,
                "gridColumnStart": 2,
                "gridRow": 1,
                "height": "98%",
                "transform": "",
              }
            }
            tabIndex="0"
          >
            <span>
              08:10
               - 
              09:50
              <br />
              <span>
                Ken
              </span>
            </span>
          </div>
        </div>
        <div
          className="calendar__rooms__cells"
          style={
            Object {
              "gridTemplateColumns": "repeat(90, 1fr)",
            }
          }
        >
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 7,
                "gridColumnStart": 1,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 13,
                "gridColumnStart": 7,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 19,
                "gridColumnStart": 13,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 25,
                "gridColumnStart": 19,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 31,
                "gridColumnStart": 25,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 37,
                "gridColumnStart": 31,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 43,
                "gridColumnStart": 37,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 49,
                "gridColumnStart": 43,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 55,
                "gridColumnStart": 49,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 61,
                "gridColumnStart": 55,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 67,
                "gridColumnStart": 61,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 73,
                "gridColumnStart": 67,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 79,
                "gridColumnStart": 73,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 85,
                "gridColumnStart": 79,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
          <div
            className="calendar__cells__cell"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "gridColumnEnd": 91,
                "gridColumnStart": 85,
                "gridRow": 1,
              }
            }
            tabIndex="0"
          />
        </div>
        <div
          className="modals"
        >
          <div
            id="reservation-details-modal"
          />
          <div
            id="reservation-details-modal"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
